libraries/chain/controller.cpp

```diff
@@ -446,8 +446,6 @@ struct controller_impl {
             if( read_mode == db_read_mode::IRREVERSIBLE ) {
                controller::block_report br;
                apply_block( br, *bitr, controller::block_status::complete, trx_meta_cache_lookup{} );
```

Comment:
shouldn't it be `controller::block_status::incomplete` so that it does `fork_db.mark_valid( bsp );` later in `commit_block()`?

Reply:
It is marked valid in `commit_block()` via:
```
         } else if (s != controller::block_status::irreversible) {
            fork_db.mark_valid( bsp );
```